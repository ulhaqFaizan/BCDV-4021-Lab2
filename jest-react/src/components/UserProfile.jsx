import React from 'react';

function UserProfile ({user}) {

    return(
        <div>
            {user ? (
                <div>
                    <h1>{user.name}</h1>
                    <p>{user.email}</p>
                </div>
            ) : (
                <p>No user info available.</p>
            )}
        </div>
    );
}

export default UserProfile;