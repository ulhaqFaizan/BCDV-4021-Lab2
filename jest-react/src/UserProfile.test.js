import React from 'react';
import { create } from 'react-test-renderer';
import UserProfile from './components/UserProfile';

it('renders correctly when there is a user', () => {
const user ={ name: 'john', email:'john@example.com'};
const tree = create(<UserProfile user={user} />).toJSON();
expect(tree).toMatchSnapshot();
});

it('renders correctly when there is no user', () => {
    const tree = create(<UserProfile />).toJSON();
    expect(tree).toMatchSnapshot();
});