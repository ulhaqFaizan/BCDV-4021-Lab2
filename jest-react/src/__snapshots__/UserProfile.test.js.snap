// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is a user 1`] = `
<div>
  <div>
    <h1>
      john
    </h1>
    <p>
      john@example.com
    </p>
  </div>
</div>
`;

exports[`renders correctly when there is no user 1`] = `
<div>
  <p>
    No user info available.
  </p>
</div>
`;
